docs/api/utils/partitionByKey.md

```diff
@@ -28,6 +28,86 @@ export function partitionByKey<T, K, R>(
 
 2. A stream with the list of active keys
 
+### Examples
+
+```ts
+const source = interval(1000);
+const [getGroupByKey, keys$] = partitionByKey(
+  source,
+  x => x % 2 == 0 ? "even" : "odd",
+  (groupedObservable$, key) => groupedObservable$.pipe(map(x => `${x} is ${key}`))
+);
+
+const [useEven, even$] = bind(getGroupByKey("even"));
+const [useOdd, odd$] = bind(getGroupByKey("odd"));
+const [useKeys] = bind(keys$);
+
+function MyComponent() {
+  const odd = useOdd();
+  const even = useEven();
+  const keys = useKeys();
+
+  return (
+    <>
+      <div>Your keys are: {keys.join(", ")}</div>
+      <div>{odd}</div>
+      <div>{even}</div>
+    </>
+  );
+}
+```
+
+A more typical list example. The list component can bind the list of keys
+while the item component binds the stream for each item, eliminating
+unnecessary renders:
+
+```tsx
+interface Pet {
+  id: number;
+  pet: string,
+  pos?: number;
+}
+
+const pets = ["Fluffy", "Bella", "Nala", "Nocturne", "Teddy"]
+               .map((pet, id): Pet => ({pet, id}));
+
+let [petRace$, petRaceDispatch] = createSignal<Pet>();
+
+// Let's line up our pets
+petRace$ = merge(of(...pets), petRace$);
+
+const [petByID, pets$] = partitionByKey(
+  petRace$,
+  x => x.id,
+)
```

Comment:
I really like this example, but I'd try to avoid changing references - How about

```suggestion
const [petUpdate$, updatePet] = createSignal<Pet>();

// Let's line up our pets
const petRace$ = merge(of(...pets), petUpdate$);

const [petByID, pets$] = partitionByKey(
  petRace$,
  x => x.id,
)
```

I'd also rename `pets$` to `petIds$` just to make it a bit more clear